registration-processor/pre-processor/registration-processor-cmd-validator-stage/src/main/java/io/mosip/registration/processor/stages/app/CMDValidationProcessor.java

```diff
@@ -146,20 +160,37 @@ public MessageDTO process(MessageDTO object, String stageName) {
 				updateAudit(description, isTransactionSuccessful, moduleId, moduleName, registrationId);
 				return object;
 			}
-
-			if (validateCenter) {
-				centerValidator.validate(primaryLanguagecode, regOsi, registrationStatusDto.getRegistrationId());
+			boolean isSubProcess=false;
+			if(subProcess!=null || !subProcess.isBlank()) {
+			 isSubProcess=Arrays.asList(subProcess.split(",")).contains(registrationStatusDto.getRegistrationType());
 			}
-
-			if (validateMachine) {
-				machineValidator.validate(regOsi.getMachineId(), primaryLanguagecode, regOsi.getPacketCreationDate(),
-						registrationStatusDto.getRegistrationId());
+			if(isSubProcess) {
+				if (validateCenter && validateCenterForSubProcess) {
```

Comment:
If there are 2 sub process 1. biometric correction which needs device validation to be done 2. document correction which does not require the device validation, how can we switch off for 1 and not for other ?